python/mlc_chat/compiler/quantization/group_quantizer.py

```diff
@@ -0,0 +1,70 @@
+"""A group quantizer for on the fly parameter quantization"""
+# pylint: disable=too-few-public-methods
+
+from typing import List, Tuple
+
+from tvm import te, tir
+
+from .quantization import QuantizeConfig
+
+
+def te_quantize(
+    weight: te.Tensor, config: QuantizeConfig
+) -> Tuple[te.Tensor, te.Tensor, List[te.Tensor]]:
+    """Group quantization for weight tensor, defined in tensor expression."""
+    # pylint: disable=too-many-locals
+    assert len(weight.shape) == 2
+    n, m = weight.shape
```

Comment:
nit: use `k` to denote the reduction axis

```suggestion
    n, k = weight.shape
```